recipes/minisat/all/conanfile.py

```diff
@@ -44,45 +36,34 @@ def configure(self):
         if self.options.shared:
             self.options.rm_safe("fPIC")
 
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
     def requirements(self):
         # https://github.com/niklasso/minisat/blob/37dc6c67e2af26379d88ce349eb9c4c6160e8543/minisat/utils/ParseUtils.h#L27
-        self.requires("zlib/1.2.13", transitive_headers=True)
+        self.requires("zlib/1.2.13", transitive_headers=True, transitive_libs=True)
 
-    def layout(self):
-        cmake_layout(self, src_folder="src")
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def generate(self):
         tc = CMakeToolchain(self)
-        tc.variables["STATIC_BINARIES"] = not self.options.shared
-        tc.variables["USE_SORELEASE"] = False
-        tc.generate()
-        tc = CMakeDeps(self)
+        tc.variables["MINISAT_SRC_DIR"] = self.source_folder.replace("\\", "/")
         tc.generate()
-
-    def _determine_lib_name(self):
-        return f"minisat-lib-{'shared' if self.options.shared else 'static'}"
+        deps = CMakeDeps(self)
+        deps.generate()
 
     def build(self):
         apply_conandata_patches(self)
         cmake = CMake(self)
-        cmake.configure()
-        cmake.build(target=self._determine_lib_name())
+        cmake.configure(build_script_folder=os.path.join(self.source_folder, os.pardir))
+        cmake.build()
 
     def package(self):
-         # Manually copy since install does both static and shared 
-         # https://github.com/conan-io/conan-center-index/pull/16071#discussion_r1149458542
-        copy(self, pattern="LICENSE", src=self.source_folder, dst=join(self.package_folder, "licenses"))
-        copy(self, pattern="*.h", src=join(self.source_folder, "minisat"), dst=join(self.package_folder, "include", "minisat"))
-        if self.options.shared:
-            copy(self, pattern="*.so*", src=self.build_folder, dst=join(self.package_folder, "lib"))
-            copy(self, pattern="*.dylib*", src=self.build_folder, dst=join(self.package_folder, "lib"))
-        else:
-            copy(self, pattern="*.a", src=self.build_folder, dst=join(self.package_folder, "lib"))
-            copy(self, pattern="*.lib", src=self.build_folder, dst=join(self.package_folder, "lib"), keep_path=False)
-        fix_apple_shared_install_name(self)
+        copy(self, pattern="LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
 
     def package_info(self):
-        self.cpp_info.libs = collect_libs(self)
-        if self.settings.os in ["Linux", "FreeBSD"]:
-            self.cpp_info.system_libs.append("m")
-            self.cpp_info.system_libs.append("pthread")
+        self.cpp_info.libs = ["minisat"]
+        self.cpp_info.includedirs.append(os.path.join("include", "minisat"))
```

Comment:
```suggestion
        self.cpp_info.includedirs.append(os.path.join("include", "minisat"))
        if self.settings.os in ["Linux", "FreeBSD"]:
            self.cpp_info.system_libs.append("m")
```

so libm is required, but not pthread